Init express

joeivans opened the pull request:
- Initialize the repo for Express js server.

joeivans committed:
Init repo starter files

joeivans committed:
Add dependencies

joeivans committed:
Add weather.json static data

joeivans committed:
Add docs/images folder

joeivans committed:
Update readme

ryanemmans commented on README.md:
```diff
@@ -0,0 +1,49 @@
+Welcome to City Explorer API! It's a simple server that connects a client to
+information about a city.
+
+## Getting Started
+<!-- What are the steps that a user must take in order to build this app on
+their own machine and get it running? -->
+Clone the source repository and run `npm install && npm run`<br />
+Visit the deployed site: [Joe's City Explorer API](https://joes-city-explorer-api.netlify.app)
+
+## Architecture
+<!-- Provide a detailed description of the application design. What technologies
+(languages, libraries, etc) you're using, and any other relevant design
+information. -->
+This is a web API server application that uses the web request/response cycle to
+interface with clients and other servers to query data and provide a client with
+high-quality results.
+
+![Architecture](./docs/images/Lab07.png)
+
+Dependencies:
+- Cors
+- Dotenv
+- Express
```
This is looking good!

SarahCreager commented on README.md:
```diff
@@ -0,0 +1,49 @@
+## Credit and Collaborations
+<!-- Give credit (and a link) to other people or resources that helped you build
+this application. -->
+- I paired with Kason Braley for the initial design of the project.
```
I like that you mentioned this. It's a great contribution to add to the README

ryanemmans commented on package.json:
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "city-explorer-api",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/joeivans/city-explorer-api.git"
+  },
+  "keywords": [],
+  "author": "Joe Ivans",
+  "license": "MIT",
+  "dependencies": {
+    "cors": "^2.8.5",
+    "dotenv": "^10.0.0",
+    "express": "^4.17.1"
```
Good job on the init setup. If you'd like, you could change index.js to server.js.

SarahCreager commented on README.md:
```diff
@@ -0,0 +1,49 @@
+Estimate of time needed to complete: 3h<br />
+Start time: 06:00<br />
+Finish time: 07:45<br />
+Actual time needed to complete: 1h45<br />
```
It's always nice when you overshoot the time estimate. Great job!

joeivans committed:
Add gitignore

joeivans committed:
Add a health-check endpoint test

joeivans committed:
Add the server

- Added the server to the solution.
- Added the PORT environment variable to the .env file outside
  of source control.
- Added a health check endpoint.
- Tested running npm start and watching the feedback.

joeivans commented:
I pushed to the same branch as yesterday

hkimbrough22 reviewed (commented):
in your data/weatherData.http I am not sure what this file is or does?

hkimbrough22 commented on server.js:
```diff
@@ -0,0 +1,50 @@
+app.use(cors());
+
+app.get('/health-check', (req, res) => {
+  console.log(Date.now());
```
Wow, this is a really cool idea. I might steal this 👍

hkimbrough22 commented on server.js:
```diff
@@ -0,0 +1,50 @@
+  res.send('healthy');
+});
+
+app.get('/photo', getPhotos);
```
Nice, you already implemented this from today's lesson!
